<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>940</width>
    <height>932</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>900</width>
    <height>850</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QWidget" name="widget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>300</width>
          <height>300</height>
         </size>
        </property>
        <layout class="QGridLayout" name="gridLayout_14">
         <item row="3" column="0">
          <widget class="QPushButton" name="save_bmp">
           <property name="text">
            <string>bmp</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QPushButton" name="save_gif">
           <property name="text">
            <string>gif</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QPushButton" name="save_jpeg">
           <property name="text">
            <string>jpeg</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="MyGLWidget" name="openGLWidget">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QGroupBox" name="groupBox_3">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>info</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="0" column="0">
             <widget class="QLabel" name="infolabel">
              <property name="text">
               <string>no body</string>
              </property>
              <property name="textFormat">
               <enum>Qt::AutoText</enum>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QScrollArea" name="scrollArea">
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>298</width>
           <height>877</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="3" column="0">
           <layout class="QGridLayout" name="gridLayout_9">
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Тип линии</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="cB_type_line">
              <item>
               <property name="text">
                <string>Сплошная</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Пунктирная</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item row="4" column="0">
           <layout class="QGridLayout" name="gridLayout_10">
            <item row="0" column="0">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>Размер</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSlider" name="horizontalSliderTypeLine">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>12</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QSpinBox" name="spinBoxTypeLine">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>12</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="6" column="0">
           <layout class="QGridLayout" name="gridLayout_12">
            <item row="0" column="2">
             <widget class="QSlider" name="horizontalSliderTypeTop">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>12</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QSpinBox" name="spinBoxTypeTop">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>12</number>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>Размер</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="5" column="0">
           <layout class="QGridLayout" name="gridLayout_11">
            <item row="0" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Тип вершины</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="comboBox_type_top">
              <item>
               <property name="text">
                <string>Круглая</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Квадратная</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Отсутствует</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <widget class="QWidget" name="gridWidget" native="true">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <layout class="QGridLayout" name="gridLayout_15">
             <item row="1" column="1">
              <widget class="QComboBox" name="ProjTypeCombo">
               <property name="currentText">
                <string>Параллельная</string>
               </property>
               <item>
                <property name="text">
                 <string>Параллельная</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Центральная</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>Тип проекции</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QPushButton" name="Open_file">
              <property name="text">
               <string>Open file</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_13">
              <item row="0" column="0">
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>Масштаб</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="ScaleSpin">
                <property name="alignment">
                 <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                </property>
                <property name="decimals">
                 <number>1</number>
                </property>
                <property name="minimum">
                 <double>0.100000000000000</double>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
                <property name="stepType">
                 <enum>QAbstractSpinBox::DefaultStepType</enum>
                </property>
                <property name="value">
                 <double>1.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string>Вращение (град)</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
              <layout class="QGridLayout" name="gridLayout_8">
               <item row="0" column="0">
                <layout class="QHBoxLayout" name="horizontalLayout">
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout">
                   <item>
                    <widget class="QLabel" name="label">
                     <property name="text">
                      <string>X</string>
                     </property>
                     <property name="alignment">
                      <set>Qt::AlignCenter</set>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="XRotateSpin">
                     <property name="maximum">
                      <number>360</number>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QDial" name="XRotateDial">
                     <property name="mouseTracking">
                      <bool>true</bool>
                     </property>
                     <property name="maximum">
                      <number>360</number>
                     </property>
                     <property name="wrapping">
                      <bool>false</bool>
                     </property>
                     <property name="notchTarget">
                      <double>3.700000000000000</double>
                     </property>
                     <property name="notchesVisible">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout_2">
                   <item>
                    <widget class="QLabel" name="label_2">
                     <property name="text">
                      <string>Y</string>
                     </property>
                     <property name="alignment">
                      <set>Qt::AlignCenter</set>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="YRotateSpin">
                     <property name="maximum">
                      <number>360</number>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QDial" name="YRotateDial">
                     <property name="maximum">
                      <number>360</number>
                     </property>
                     <property name="notchesVisible">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout_3">
                   <item>
                    <widget class="QLabel" name="label_3">
                     <property name="text">
                      <string>Z</string>
                     </property>
                     <property name="alignment">
                      <set>Qt::AlignCenter</set>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="ZRotateSpin">
                     <property name="maximum">
                      <number>360</number>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QDial" name="ZRotateDial">
                     <property name="maximum">
                      <number>360</number>
                     </property>
                     <property name="notchesVisible">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox_2">
              <property name="title">
               <string>Перемещение</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
              <layout class="QGridLayout" name="gridLayout_4">
               <item row="3" column="0">
                <widget class="QSplitter" name="splitter_4">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <widget class="QLabel" name="label_12">
                  <property name="text">
                   <string>Z</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignCenter</set>
                  </property>
                 </widget>
                 <widget class="QSlider" name="ZTranslateSlider">
                  <property name="minimum">
                   <number>-100</number>
                  </property>
                  <property name="maximum">
                   <number>100</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="invertedAppearance">
                   <bool>false</bool>
                  </property>
                  <property name="invertedControls">
                   <bool>false</bool>
                  </property>
                 </widget>
                 <widget class="QSpinBox" name="ZTranslateSpin">
                  <property name="minimum">
                   <number>-100</number>
                  </property>
                  <property name="maximum">
                   <number>100</number>
                  </property>
                 </widget>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QSplitter" name="splitter_2">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <widget class="QLabel" name="label_10">
                  <property name="text">
                   <string>X</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignCenter</set>
                  </property>
                 </widget>
                 <widget class="QSlider" name="XTranslateSlider">
                  <property name="minimum">
                   <number>-100</number>
                  </property>
                  <property name="maximum">
                   <number>100</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                 <widget class="QSpinBox" name="XTranslateSpin">
                  <property name="minimum">
                   <number>-100</number>
                  </property>
                  <property name="maximum">
                   <number>100</number>
                  </property>
                 </widget>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QSplitter" name="splitter_3">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <widget class="QLabel" name="label_11">
                  <property name="text">
                   <string>Y</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignCenter</set>
                  </property>
                 </widget>
                 <widget class="QSlider" name="YTranslateSlider">
                  <property name="minimum">
                   <number>-100</number>
                  </property>
                  <property name="maximum">
                   <number>100</number>
                  </property>
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                 </widget>
                 <widget class="QSpinBox" name="YTranslateSpin">
                  <property name="minimum">
                   <number>-100</number>
                  </property>
                  <property name="maximum">
                   <number>100</number>
                  </property>
                 </widget>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox_5">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>100</height>
               </size>
              </property>
              <property name="title">
               <string>Настройки цвета</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
              <layout class="QGridLayout" name="gridLayout_5">
               <item row="1" column="0">
                <layout class="QGridLayout" name="gridLayout_6">
                 <item row="0" column="0">
                  <widget class="QPushButton" name="pBColorTop">
                   <property name="text">
                    <string>Вершина</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QPushButton" name="pBColorLine">
                   <property name="text">
                    <string>Линия</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="2">
                  <widget class="QPushButton" name="pBBackground">
                   <property name="text">
                    <string>Фон</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionopen">
   <property name="text">
    <string>open</string>
   </property>
  </action>
  <action name="actionclose">
   <property name="text">
    <string>close</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>myglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
